### PR TITLE
Initialize class/struct variables with default values in modules/

### PR DESCRIPTION
**Pull request by qarmin**
Part of #43636

In future it may also allow to run Memory sanitizer without too big error spam

### CODE REVIEW

File: modules/enet/networked_multiplayer_enet.cpp [hunk 1]
```diff
@@ -883,6 +883,8 @@ NetworkedMultiplayerENet::NetworkedMultiplayerENet() {
 	enet_compressor.compress = enet_compress;
 	enet_compressor.decompress = enet_decompress;
 	enet_compressor.destroy = enet_compressor_destroy;
+	peer = nullptr;
+	host = nullptr;
```
Comment (Faless) [inline modules/enet/networked_multiplayer_enet.cpp:887]: shouldn't this be done in the header file?

File: modules/webrtc/webrtc_data_channel_js.h [hunk 2]
```diff
@@ -42,16 +42,16 @@ class WebRTCDataChannelJS : public WebRTCDataChannel {
 	String _label;
 	String _protocol;
 
-	bool _was_string;
+	bool _was_string = false;
 	WriteMode _write_mode;
```
Comment (Faless) [inline modules/webrtc/webrtc_data_channel_js.h:46]: why not this one too?

File: modules/webrtc/webrtc_data_channel_js.cpp [hunk 3]
```diff
@@ -185,7 +185,6 @@ WebRTCDataChannelJS::WebRTCDataChannelJS() {
 	queue_count = 0;
 	_was_string = false;
 	_write_mode = WRITE_MODE_BINARY;
-	_js_id = 0;
```
Comment (Faless) [inline modules/webrtc/webrtc_data_channel_js.cpp:188]: Why remove this, but not the others? i.e. queue_count, _was_string, etc?

File: modules/text_server_adv/icu_data/icudata_stub.cpp [hunk 4]
```diff
@@ -33,12 +33,12 @@
...
-	uint8_t magic_1, magic_2;
+	uint16_t header_size = 0;
+	uint8_t magic_1 = 0, magic_2 = 0;
 	UDataInfo info;
-	char padding[8];
-	uint32_t count, reserved;
-	int fake_name_and_data[4];
+	char padding[8] = {};
+	uint32_t count = 0, reserved = 0;
+	int fake_name_and_data[4] = {};
```
Comment (bruvzg) [inline modules/text_server_adv/icu_data/icudata_stub.cpp:41]: This one is only used to define `U_ICUDATA_ENTRY_POINT` on the next line.

File: modules/text_server_adv/script_iterator.h [hunk 5]
```diff
@@ -45,8 +45,8 @@
 class ScriptIterator {
 public:
 	struct ScriptRange {
-		int start;
-		int end;
+		int start = 0;
+		int end = 0;
 		hb_script_t script;
```
Comment (bruvzg) [inline modules/text_server_adv/script_iterator.h:50]:
```suggestion
		hb_script_t script = HB_SCRIPT_COMMON;
```

File: modules/text_server_fb/dynamic_font_fb.h [hunk 6]
```diff
@@ -70,14 +70,11 @@ struct DynamicFontDataFallback : public FontDataFallback {
...
-			uint32_t key;
+			uint32_t key = 0;
 		};
 		bool operator<(CacheID right) const {
 			return key < right.key;
 		}
-		CacheID() {
-			key = 0;
-		}
 	};
```
Comment (bruvzg) [inline modules/text_server_fb/dynamic_font_fb.h:78]: The same should be done in the `modules/text_server_adv/dynamic_font_adv.h`.

File: modules/gdnative/gdnative_library_editor_plugin.cpp [hunk 7]
```diff
@@ -414,6 +414,8 @@ GDNativeLibraryEditorPlugin::GDNativeLibraryEditorPlugin(EditorNode *p_node) {
 	library_editor->set_custom_minimum_size(Size2(0, 250 * EDSCALE));
 	button = p_node->add_bottom_panel_item(TTR("GDNativeLibrary"), library_editor);
 	button->hide();
+
+	editor = nullptr;
```
Comment (akien-mga) [inline modules/gdnative/gdnative_library_editor_plugin.cpp:418]: Shouldn't this go in the header? And `button` and `library_editor` can also be initialized to `nullptr` to be safe, even if they then get assigned values here.

File: modules/mobile_vr/mobile_vr_interface.cpp [hunk 8]
```diff
@@ -448,19 +448,7 @@ void MobileVRInterface::process() {
...
-	eye_height = 1.85;
-	intraocular_dist = 6.0;
-	display_width = 14.5;
-	display_to_lens = 4.0;
-	oversample = 1.5;
-	k1 = 0.215;
-	k2 = 0.215;
-	last_ticks = 0;
-};
+MobileVRInterface::MobileVRInterface(){};
```
Comment (akien-mga) [inline modules/mobile_vr/mobile_vr_interface.cpp:451]:
```suggestion
MobileVRInterface::MobileVRInterface() {}
```

File: modules/mobile_vr/mobile_vr_interface.cpp [hunk 9]
```diff
@@ -448,19 +448,7 @@ void MobileVRInterface::process() {
 	};
 };
 
-MobileVRInterface::MobileVRInterface() {
-	initialized = false;
-
-	// Just set some defaults for these. At some point we need to look at adding a lookup table for common device + headset combos and/or support reading cardboard QR codes
```
Comment (akien-mga) [inline modules/mobile_vr/mobile_vr_interface.cpp:454]: Would be good to port this comment to the header too.

File: modules/stb_vorbis/audio_stream_ogg_vorbis.h [hunk 10]
```diff
@@ -41,11 +41,11 @@ class AudioStreamOGGVorbis;
 class AudioStreamPlaybackOGGVorbis : public AudioStreamPlaybackResampled {
```
Comment (akien-mga) [inline modules/stb_vorbis/audio_stream_ogg_vorbis.h:41]: You missed `AudioStreamOGGVorbis()` which has plenty of stuff to move to the header.

File: modules/websocket/wsl_peer.h [hunk 11]
```diff
@@ -48,29 +48,17 @@ class WSLPeer : public WebSocketPeer {
...
+		bool valid = false;
+		bool is_server = false;
+		bool closing = false;
+		void *obj = nullptr;
+		void *peer = nullptr;
 		Ref<StreamPeer> conn;
 		Ref<StreamPeerTCP> tcp;
-		int id;
+		int id = 1;
 		wslay_event_context_ptr ctx;
```
Comment (akien-mga) [inline modules/websocket/wsl_peer.h:61]:
```suggestion
		wslay_event_context_ptr ctx = nullptr;
```

**Comment by akien-mga** [comment]
Left a few comments, otherwise should be good to merge, I looked through the whole diff.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**